testFramework/com/twitter/intellij/pants/testFramework/PantsIntegrationTestCase.java

```diff
@@ -495,6 +499,21 @@ public void tearDown() throws Exception {
       killNailgun();
       cleanProjectRoot();
       Messages.setTestDialog(TestDialog.DEFAULT);
+
+      // TODO(dmcclanahan): after updating from 172.4343.14 to 173.3531.6,
+      // intellij's provided test class sometimes yells about a leaky jdk
+      // table. i don't think this indicates any problems, so for now if tests
+      // fail with leaky sdk errors, broaden this to include the leaked sdks
+      final Stream<Sdk> leakedJdks = Arrays.stream(ProjectJdkTable.getInstance().getAllJdks())
+        .filter(jdk -> {
+            final String name = jdk.getName();
+            return name.contains("pants") || name.startsWith("python");
```

Comment:
are you checking for JDK only or also python interpreter?

Reply:
Running the `tests:misc` target with `--test-junit-test=com.twitter.intellij.pants.execution.OSSPantsPythonRunConfigurationIntegrationTest` fails with a leaky jdk table error, and the "leaked" sdk is just named `python:  Python 2.7 (/usr/bin/python)`.

I don't like this, but it seems like the best solution right now. If we expand the number of python imports we do in integration testing I would feel less comfortable, but for now I think this is the only instance. I have made this less of a concern with the most recent commit, only checking for `.startsWith("python")` in `OSSPantsPythonRunConfigurationIntegrationTest#tearDown`.